src/wallet/rpcnames.cpp

```diff
@@ -823,6 +825,151 @@ name_update ()
 
 /* ************************************************************************** */
 
+RPCHelpMan
+queuerawtransaction ()
+{
+  return RPCHelpMan ("queuerawtransaction",
+      "\nQueue a transaction for future broadcast.",
+      {
+          {"hexstring", RPCArg::Type::STR, RPCArg::Optional::NO, "The hex string of the raw transaction"},
+      },
+      RPCResult {RPCResult::Type::STR_HEX, "", "the transaction ID"},
+      RPCExamples {
+          HelpExampleCli("queuerawtransaction", "txhex") +
+          HelpExampleRpc("queuerawtransaction", "txhex")
+      },
+      [&] (const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+  std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest (request);
+  if (!wallet) return NullUniValue;
+  CWallet* const pwallet = wallet.get ();
+
+  RPCTypeCheck (request.params,
+                {UniValue::VSTR});
+
+  // parse transaction from parameter
+  CMutableTransaction mtxParsed;
+  if (!DecodeHexTx(mtxParsed, request.params[0].get_str(), true, true))
+    throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+  CTransactionRef txParsed(MakeTransactionRef(mtxParsed));
+  const uint256& hashTx = txParsed->GetHash();
+
+  // Validate transaction
+  NodeContext& node = EnsureAnyNodeContext(request.context);
+  ChainstateManager& chainman = EnsureChainman(node);
+  CTxMemPool& mempool = EnsureMemPool(node);
+
+  LOCK2 (pwallet->cs_wallet, cs_main);
+  MempoolAcceptResult result = AcceptToMemoryPool(chainman.ActiveChainstate(), mempool, txParsed,
+    /* bypass_limits */ false, /* test_accept */ true);
+  if (result.m_result_type != MempoolAcceptResult::ResultType::VALID &&
+      result.m_state.GetResult() == TxValidationResult::TX_CONSENSUS)
+      /* We only want to avoid unconditionally invalid transactions.
+       * Blocking e.g. orphan transactions is not desirable. */
+    throw JSONRPCError (RPC_WALLET_ERROR, strprintf("Invalid transaction (%s)", result.m_state.GetRejectReason()));
```

Comment:
Is my understanding here correct, that queueing a transaction basically means "send it at any time it is possible"?  So if it is completely valid right now, it will just be sent into the mempool and mined right away.  But if it is not valid due to a missing input or something, then this check will still succeed, but will not put it into the mempool, and it will be queued instead?

In case the accept is completely valid, should it be written to the queue at all?  Maybe in this case, the same as `sendrawtransaction` should happen instead (i.e. it is added to the wallet as normal transaction and broadcast directly)?

Reply:
Yeah. I think there's two ways of doing this; either manually check (we have the `TxValidationResult` right there, so it's not exactly hard), or just `EffectTransactionQueue` after each add. I think I prefer the former, since it seems like it could potentially cause trouble to call into itself like that, but I might just be superstitious. The latter would be slightly less code.

Reply:
Yeah, I think the first approach is probably cleaner (basically you would never even touch the tx queue, instead of adding and then removing again), assuming it is not too much extra code compared to the second.